### PR TITLE
[ENH][BF] Error handling for IVIM VarPro NLLS 

### PR DESCRIPTION
**Pull request by ShreyasFadnavis**
- [x] Fixes the issue #1828

### CODE REVIEW

**Comment by ShreyasFadnavis** [comment]
@arokem can you take a look at this PR? I think its a minor fix to something that was missing in the VarPro implementation..

File: dipy/reconst/ivim.py [hunk 1]
```diff
@@ -622,6 +619,49 @@ def fit(self, data, bounds_de=None):
...
+                                x_f,
+                                bounds=bounds,
+                                args=(data,))
+            ivim_params = res.x
+            if np.all(np.isnan(ivim_params)):
+                return np.array([-1, -1, -1, -1])
+            return ivim_params
+
+        except ValueError:
+            ivimmodel = IvimModel(gtab, fit_method='trr')
```
Comment (arokem) [inline dipy/reconst/ivim.py]: This quietly falls back to the `trr` method? I am not sure that's what we want here. Imagine a user comparing varpro to trr and getting exactly the same for both? Is there some way to instead throw a sufficiently informative error here?

Comment (ShreyasFadnavis) [reply dipy/reconst/ivim.py]: I understand what you mean, I could alternatively return the values from `x_f` obtained from `convex optimization + global optimization` of `VarPro` step too. 

But, this `x0` error typically occurs when the mask is not applied and is working in the background and not the data. 
So I feel that this should be okay.. Does that make sense?  

Nevertheless, I you feel that returning values from the previous step is better, I can do that too.. Just need to think of an appropriate way to do that.

Comment (ShreyasFadnavis) [reply dipy/reconst/ivim.py]: @Garyfallidis which one of the 2 alternatives do you suggest?

**Comment by ShreyasFadnavis** [comment]
PR on wrong branch! Will create a new one! Sorry..

**Pull request closed without merging**